common.js

```diff
@@ -16,7 +16,7 @@ console.log()
 
 async function downloadData() {
 
-    let uri = serverConfig.dataBaseUri + '/csv';
+    let uri = serverConfig.dataBaseUri;
```

Comment:
This change should end with a modification in the ```config.json``` file in order to ask for the ```...uri/csv``` data
For now, only a Forbidden error occures

here it is:

```javascript
db update scheduled to run at Thu Apr 02 2020 18:25:30 GMT+0200 (Central European Summer Time) started at Thu Apr 02 2020 18:25:30 GMT+0200 (Central European Summer Time)
getting https://opendata.ecdc.europa.eu/covid19/casedistribution
error when getting https://opendata.ecdc.europa.eu/covid19/casedistribution
Error: Forbidden
    at Request.callback (/home/digitalvertices/cLab/visualisation-cas-covid-19/node_modules/superagent/lib/node/index.js:871:15)
    at IncomingMessage.<anonymous> (/home/digitalvertices/cLab/visualisation-cas-covid-19/node_modules/superagent/lib/node/index.js:1114:20)
    at Stream.emit (events.js:305:20)
    at Unzip.<anonymous> (/home/digitalvertices/cLab/visualisation-cas-covid-19/node_modules/superagent/lib/node/unzip.js:53:12)
    at Unzip.emit (events.js:317:22)
    at endReadableNT (_stream_readable.js:1215:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  status: 403,
  response: <ref *1> Response {
    _events: [Object: null prototype] {},
    _eventsCount: 0,
    _maxListeners: undefined,
    res: IncomingMessage {
      _readableState: [ReadableState],
      readable: false,
      _events: [Object: null prototype],
      _eventsCount: 3,
      _maxListeners: undefined,
      socket: [TLSSocket],
      httpVersionMajor: 1,
      httpVersionMinor: 1,
      httpVersion: '1.1',
      complete: true,
      headers: [Object],
      rawHeaders: [Array],
      trailers: {},
      rawTrailers: [],
      aborted: false,
      upgrade: false,
      url: '',
      method: null,
      statusCode: 403,
      statusMessage: 'Forbidden',
      client: [TLSSocket],
      _consuming: false,
      _dumped: false,
      req: [ClientRequest],
      setEncoding: [Function (anonymous)],
      on: [Function (anonymous)],
      text: '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">\r\n' +
        '<html xmlns="http://www.w3.org/1999/xhtml">\r\n' +
        '<head>\r\n' +
        '<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1"/>\r\n' +
        '<title>403 - Forbidden: Access is denied.</title>\r\n' +
        '<style type="text/css">\r\n' +
        '<!--\r\n' +
        'body{margin:0;font-size:.7em;font-family:Verdana, Arial, Helvetica, sans-serif;background:#EEEEEE;}\r\n' +
        'fieldset{padding:0 15px 10px 15px;} \r\n' +
        'h1{font-size:2.4em;margin:0;color:#FFF;}\r\n' +
        'h2{font-size:1.7em;margin:0;color:#CC0000;} \r\n' +
        'h3{font-size:1.2em;margin:10px 0 0 0;color:#000000;} \r\n' +
        '#header{width:96%;margin:0 0 0 0;padding:6px 2% 6px 2%;font-family:"trebuchet MS", Verdana, sans-serif;color:#FFF;\r\n' +
        'background-color:#555555;}\r\n' +
        '#content{margin:0 0 0 2%;position:relative;}\r\n' +
        '.content-container{background:#FFF;width:96%;margin-top:8px;padding:10px;position:relative;}\r\n' +
        '-->\r\n' +
        '</style>\r\n' +
        '</head>\r\n' +
        '<body>\r\n' +
        '<div id="header"><h1>Server Error</h1></div>\r\n' +
        '<div id="content">\r\n' +
        ' <div class="content-container"><fieldset>\r\n' +
        '  <h2>403 - Forbidden: Access is denied.</h2>\r\n' +
        '  <h3>You do not have permission to view this directory or page using the credentials that you supplied.</h3>\r\n' +
        ' </fieldset></div>\r\n' +
        '</div>\r\n' +
        '</body>\r\n' +
        '</html>\r\n',
      [Symbol(kCapture)]: false
    },
    request: Request {
      _events: [Object: null prototype],
      _eventsCount: 1,
      _maxListeners: undefined,
      _enableHttp2: false,
      _agent: false,
      method: 'GET',
      url: 'https://opendata.ecdc.europa.eu/covid19/casedistribution/',
      _header: [Object],
      header: [Object],
      writable: true,
      _redirects: 1,
      _maxRedirects: 5,
      cookies: '',
      qs: {},
      _query: [],
      qsRaw: [],
      _redirectList: [Array],
      _streamRequest: false,
      _buffer: true,
      _disableTLSCerts: true,
      protocol: 'https:',
      host: 'opendata.ecdc.europa.eu',
      _endCalled: true,
      _callback: [Function (anonymous)],
      _fullfilledPromise: [Promise],
      res: [IncomingMessage],
      _data: null,
      req: [ClientRequest],
      _resBuffered: true,
      response: [Circular *1],
      called: true,
      [Symbol(kCapture)]: false
    },
    req: ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 3,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      useChunkedEncodingByDefault: false,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      _contentLength: 0,
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      socket: [TLSSocket],
      _header: 'GET /covid19/casedistribution/ HTTP/1.1\r\n' +
        'Host: opendata.ecdc.europa.eu\r\n' +
        'accept-encoding: gzip, deflate\r\n' +
        'accept: application/octet-stream\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _onPendingData: [Function: noopPendingOutput],
      agent: [Agent],
      socketPath: undefined,
      method: 'GET',
      maxHeaderSize: undefined,
      path: '/covid19/casedistribution/',
      _ended: true,
      res: [IncomingMessage],
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    text: '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">\r\n' +
      '<html xmlns="http://www.w3.org/1999/xhtml">\r\n' +
      '<head>\r\n' +
      '<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1"/>\r\n' +
      '<title>403 - Forbidden: Access is denied.</title>\r\n' +
      '<style type="text/css">\r\n' +
      '<!--\r\n' +
      'body{margin:0;font-size:.7em;font-family:Verdana, Arial, Helvetica, sans-serif;background:#EEEEEE;}\r\n' +
      'fieldset{padding:0 15px 10px 15px;} \r\n' +
      'h1{font-size:2.4em;margin:0;color:#FFF;}\r\n' +
      'h2{font-size:1.7em;margin:0;color:#CC0000;} \r\n' +
      'h3{font-size:1.2em;margin:10px 0 0 0;color:#000000;} \r\n' +
      '#header{width:96%;margin:0 0 0 0;padding:6px 2% 6px 2%;font-family:"trebuchet MS", Verdana, sans-serif;color:#FFF;\r\n' +
      'background-color:#555555;}\r\n' +
      '#content{margin:0 0 0 2%;position:relative;}\r\n' +
      '.content-container{background:#FFF;width:96%;margin-top:8px;padding:10px;position:relative;}\r\n' +
      '-->\r\n' +
      '</style>\r\n' +
      '</head>\r\n' +
      '<body>\r\n' +
      '<div id="header"><h1>Server Error</h1></div>\r\n' +
      '<div id="content">\r\n' +
      ' <div class="content-container"><fieldset>\r\n' +
      '  <h2>403 - Forbidden: Access is denied.</h2>\r\n' +
      '  <h3>You do not have permission to view this directory or page using the credentials that you supplied.</h3>\r\n' +
      ' </fieldset></div>\r\n' +
      '</div>\r\n' +
      '</body>\r\n' +
      '</html>\r\n',
    body: {},
    files: undefined,
    buffered: true,
    headers: {
      'content-type': 'text/html',
      date: 'Thu, 02 Apr 2020 16:25:28 GMT',
      connection: 'close',
      'content-length': '737',
      'set-cookie': [Array],
      'strict-transport-security': 'max-age=16070400; includeSubDomains',
      vary: 'Accept-Encoding',
      'content-encoding': 'gzip'
    },
    header: {
      'content-type': 'text/html',
      date: 'Thu, 02 Apr 2020 16:25:28 GMT',
      connection: 'close',
      'content-length': '737',
      'set-cookie': [Array],
      'strict-transport-security': 'max-age=16070400; includeSubDomains',
      vary: 'Accept-Encoding',
      'content-encoding': 'gzip'
    },
    statusCode: 403,
    status: 403,
    statusType: 4,
    info: false,
    ok: false,
    redirect: false,
    clientError: true,
    serverError: false,
    error: Error: cannot GET /covid19/casedistribution/ (403)
        at Response.toError (/home/digitalvertices/cLab/visualisation-cas-covid-19/node_modules/superagent/lib/node/response.js:98:13)
        at ResponseBase._setStatusProperties (/home/digitalvertices/cLab/visualisation-cas-covid-19/node_modules/superagent/lib/response-base.js:119:48)
        at new Response (/home/digitalvertices/cLab/visualisation-cas-covid-19/node_modules/superagent/lib/node/response.js:44:8)
        at Request._emitResponse (/home/digitalvertices/cLab/visualisation-cas-covid-19/node_modules/superagent/lib/node/index.js:918:18)
        at IncomingMessage.<anonymous> (/home/digitalvertices/cLab/visualisation-cas-covid-19/node_modules/superagent/lib/node/index.js:1114:42)
        at Stream.emit (events.js:305:20)
        at Unzip.<anonymous> (/home/digitalvertices/cLab/visualisation-cas-covid-19/node_modules/superagent/lib/node/unzip.js:53:12)
        at Unzip.emit (events.js:317:22)
        at endReadableNT (_stream_readable.js:1215:12)
        at processTicksAndRejections (internal/process/task_queues.js:84:21) {
      status: 403,
      text: '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">\r\n' +
        '<html xmlns="http://www.w3.org/1999/xhtml">\r\n' +
        '<head>\r\n' +
        '<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1"/>\r\n' +
        '<title>403 - Forbidden: Access is denied.</title>\r\n' +
        '<style type="text/css">\r\n' +
        '<!--\r\n' +
        'body{margin:0;font-size:.7em;font-family:Verdana, Arial, Helvetica, sans-serif;background:#EEEEEE;}\r\n' +
        'fieldset{padding:0 15px 10px 15px;} \r\n' +
        'h1{font-size:2.4em;margin:0;color:#FFF;}\r\n' +
        'h2{font-size:1.7em;margin:0;color:#CC0000;} \r\n' +
        'h3{font-size:1.2em;margin:10px 0 0 0;color:#000000;} \r\n' +
        '#header{width:96%;margin:0 0 0 0;padding:6px 2% 6px 2%;font-family:"trebuchet MS", Verdana, sans-serif;color:#FFF;\r\n' +
        'background-color:#555555;}\r\n' +
        '#content{margin:0 0 0 2%;position:relative;}\r\n' +
        '.content-container{background:#FFF;width:96%;margin-top:8px;padding:10px;position:relative;}\r\n' +
        '-->\r\n' +
        '</style>\r\n' +
        '</head>\r\n' +
        '<body>\r\n' +
        '<div id="header"><h1>Server Error</h1></div>\r\n' +
        '<div id="content">\r\n' +
        ' <div class="content-container"><fieldset>\r\n' +
        '  <h2>403 - Forbidden: Access is denied.</h2>\r\n' +
        '  <h3>You do not have permission to view this directory or page using the credentials that you supplied.</h3>\r\n' +
        ' </fieldset></div>\r\n' +
        '</div>\r\n' +
        '</body>\r\n' +
        '</html>\r\n',
      method: 'GET',
      path: '/covid19/casedistribution/'
    },
    created: false,
    accepted: false,
    noContent: false,
    badRequest: false,
    unauthorized: false,
    notAcceptable: false,
    forbidden: true,
    notFound: false,
    unprocessableEntity: false,
    type: 'text/html',
    links: {},
    setEncoding: [Function: bound ],
    redirects: [ 'https://opendata.ecdc.europa.eu/covid19/casedistribution/' ],
    pipe: [Function (anonymous)],
    [Symbol(kCapture)]: false
  }
}
(node:6312) UnhandledPromiseRejectionWarning: TypeError: records is not iterable
    at getByCountry (/home/digitalvertices/cLab/visualisation-cas-covid-19/db/updateDb.js:13:24)
```

Reply:
(error just for fun ^^)

Reply:
yeah i fixed that already ! ^^ :muscle: